### PR TITLE
create new job and adding applicants

### PR DESCRIPTION
**Pull request by sermolenko-bmt**
- added service for sending email
- added templates for emails
- added endpoint for adding new applicants
- move functions for saving fails to new file

### CODE REVIEW

File: api/emailTemplates/addJob.ejs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<p>User with <strong><%=user.id%></strong> id made a request to add job.</p>
```
Comment (zaikin-andrew) [inline api/emailTemplates/addJob.ejs]: You have much data about the user. Could you add `firstName`, `lastName`, `email`?

File: api/emailTemplates/addJob.ejs [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+<p>Job have name: <strong><%=jobName%></strong></p>
+<p>Request to add to an customer with: <strong><%=customerId%></strong> id</p>
```
Comment (zaikin-andrew) [inline api/emailTemplates/addJob.ejs]: We are building a useful service. Let's add the name of the customer also.

File: api/services/uploadFiles.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+module.exports = {
+  uploadFile,
+  getFilesPath,
+  parseCSV,
+  getFolderName
+};
+
+async function uploadFile(file) {
+  return new Promise((resolve, reject) => {
+    file.upload((err, uploadedFiles) => {
```
Comment (zaikin-andrew) [inline api/services/uploadFiles.js]: try to use `util.promisify` https://nodejs.org/api/util.html#util_util_promisify_original

File: api/services/uploadFiles.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      const dirList = await fs.ensureDir(folder)
+        .then(() => decompress(filePath, folder))
+        .then(() => fs.readdir(folder))
+        .then(list => getAbsolutePaths(list, folder));
+      return Promise.resolve(dirList.filter(item => !/CollectorList\.csv/.test(item) && path.extname(item) === '.csv'));
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  }
+  return Promise.reject(new Error({ message: 'Wrong file type' }));
```
Comment (zaikin-andrew) [inline api/services/uploadFiles.js]: static string. move to `Messages`

File: api/services/uploadFiles.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  //   const macFolder = `${folder}/${dirList.filter(item => item !== '__MACOSX')[0]}`;
+  //   return fs.readdir(macFolder)
+  //     .then(items => items.map(item => `${macFolder}/${item}`));
+  // }
+  return dirList.filter(item => item !== '__MACOSX').map(item => `${folder}/${item}`);
+}
+
+
+function parseCSV(data) {
+  return new Promise((res, rej) => {
```
Comment (zaikin-andrew) [inline api/services/uploadFiles.js]: try to use `util.promisify` https://nodejs.org/api/util.html#util_util_promisify_original

File: api/emailTemplates/addApplicants.ejs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<p>User with <strong><%=user.id%></strong> id made a request to add applicants.</p>
```
Comment (zaikin-andrew) [inline api/emailTemplates/addApplicants.ejs]: You have much data about the user. Could you add `firstName`, `lastName`, `email`?

File: api/emailTemplates/addApplicants.ejs [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+<p>Request to add to an job with: <strong><%=jobId%></strong> id</p>
```
Comment (zaikin-andrew) [inline api/emailTemplates/addApplicants.ejs]: We are building a useful service. Let's add the name of the job also.

File: api/services/uploadFiles.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
+const fs = require('fs-extra');
```
Comment (zaikin-andrew) [inline api/services/uploadFiles.js]: Let's make all services' filenames capitalized, ok?

File: api/services/JobService.js [hunk 9]
```diff
@@ -224,6 +232,27 @@ async function updateQuestionById(jobId, file) {
   return addOrUpdateQuestionsForJobById(jobId, file, true);
 }
 
+async function addApplicantsById(jobId, file, stringApplicants, user) {
+  try {
+    const applicants = typeof stringApplicants === 'string' ? JSON.parse(stringApplicants.applicants) : null;
+    // get path file
+    let pathToFile;
+    const files = await uploadFiles.uploadFile(file);
```
Comment (zaikin-andrew) [inline api/services/JobService.js]: Let's check if the file exists earlier and if `not` just skip all operations with it.

Comment (sermolenko-bmt) [reply api/services/JobService.js]: we do so, file - this is stream, first we need upload file and close stream then check all files

File: config/connections.js [hunk 10]
```diff
@@ -40,31 +40,24 @@ module.exports.connections = {
...
-  // },
+
   developmentMysql: {
     adapter: 'sails-mysql',
     host: '127.0.0.1',
     port: '3306',
-    user: 'root', // optional
-    password: 'Hirefy_DB', // optional
-    database: 'dbhirefydev', // optional
+    user: process.env.MYSQL_DEV_USER, // optional
```
Comment (zaikin-andrew) [inline config/connections.js]: leave values for the development as default

**Pull request merged**